src/cargo/util/progress.rs

```diff
@@ -74,6 +74,113 @@ struct Format {
     style: ProgressStyle,
     max_width: usize,
     max_print: usize,
+    taskbar: TaskbarProgress,
+}
+
+/// Taskbar progressbar
+///
+/// Outputs ANSI sequences according to the `Operating system commands`.
+struct TaskbarProgress {
+    enabled: bool,
+    error: bool,
+}
+
+/// A taskbar progress value printable as ANSI OSC escape code
+enum TaskbarValue {
+    /// Do not output anything
+    None,
+    /// Remove progress
+    Remove,
+    /// Progress value 0-100
+    Value(f64),
+    /// Indeterminate state (no bar, just animation)
+    Indeterminate,
+    /// Progress value 0-100 in the error state
+    Error(f64),
+}
+
+enum ProgressOutput {
+    /// Print progress without a message
+    PrintNow,
+    /// Progress, message and taskbar progress
+    TextAndTaskbar(String, TaskbarValue),
+    /// Only taskbar progress, no message and no text progress
+    Taskbar(TaskbarValue),
+}
+
+impl TaskbarProgress {
+    #[cfg(test)]
+    fn new(enabled: bool) -> Self {
+        Self {
+            enabled,
+            error: false,
+        }
+    }
+
+    /// Creates a new `TaskbarProgress` from a cargo's config.
+    /// If not explicitly enabled or disabled, detect a supported terminal.
+    fn from_config(gctx: &GlobalContext) -> Self {
+        let enabled = match gctx.progress_config().taskbar {
+            Some(v) => v,
+            None => {
+                gctx.get_env("WT_SESSION").is_ok()
```

Comment:
Could we have a comment for the source of how each env var works for auto-detection?

Reply:
Done. Also replaced the match block with `unwrap_or_else()`